### PR TITLE
Setup redux toolkit

### PR DESCRIPTION
**Pull request by sarikasoni12**

### CODE REVIEW

File: echo
Comment (brenelz) [inline echo:1]: I think you created this file by mistake?

File: src/cart/cartSlice.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React  from "react";
+import { createSlice } from "@reduxjs/toolkit";
+
+const cartSlice = createSlice({
```
Comment (brenelz) [inline src/cart/cartSlice.js:4]: nice... how did you find using this?

File: src/products/components/ProductsListing.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Currency from '../../util/currency';
+import cartSlice from '../../cart/cartSlice';
+import { fetchAllProducts } from '../productsSlice';
+
+const ProductsListing = () => { 
+  const {addToCart, removeFromCart} = cartSlice.actions
+  const state = useSelector(state => state)
+  const {products} = state
```
Comment (brenelz) [inline src/products/components/ProductsListing.js]: you could combine these 2 lines

File: src/products/components/ProductsListing.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchAllProducts('http://localhost:3000/products'))
```
Comment (brenelz) [inline src/products/components/ProductsListing.js:13]: could fetchAllProducts be an action too?

File: src/cart/cartSlice.js (hunk 1 above)
Comment (sarikasoni12) [reply src/cart/cartSlice.js:4]: > nice... how did you find using this?

It's very neat and clearly separates the reducers of individual modules which helps in organizing the state management in better way

**Pull request merged**